src/ui/public/directives/__tests__/auto_select_if_only_one.js

```diff
@@ -48,4 +48,13 @@ describe('Auto-select if only one directive', function () {
 
     expect($rootScope.value).to.be(multiOptions[1]);
   });
+
+  it('should auto-select if the collection changes', function () {
+    $rootScope.options = multiOptions;
+    $rootScope.$digest();
+    expect($rootScope.value).to.not.be.ok();
+    $rootScope.options = oneOption;
```

Comment:
Can you add a test that checks what happens when `$rootScope.options` is mutated (rather than replaced)